src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -73,6 +74,28 @@ module.exports = function(scheduleUpdate : (fiber: Fiber, priorityLevel : Priori
     },
   };
 
+  function checkShouldComponentUpdate(workInProgress, oldProps, newProps, newState) {
+    if (oldProps === null ||
+        (workInProgress.updateQueue && workInProgress.updateQueue.isForced)) {
+      return true;
+    }
+
+    const instance = workInProgress.stateNode;
+    if (typeof instance.shouldComponentUpdate === 'function') {
+      return instance.shouldComponentUpdate(newProps, newState);
+    }
+
+    const type = workInProgress.type;
+    if (type.prototype && type.prototype.isPureReactComponent) {
```

Comment:
We could add a new TypeOfWork https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactTypeOfWork.js